kbs/src/api/src/policy_engine/opa/mod.rs

```diff
@@ -2,43 +2,25 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::policy_engine::PolicyEngineInterface;
-use anyhow::{anyhow, Result};
+use crate::policy_engine::{PolicyEngineInterface, ResourcePolicyError};
 use async_trait::async_trait;
 use base64::Engine;
-use serde_json::Value;
-use std::ffi::CStr;
 use std::fs;
-use std::os::raw::c_char;
 use std::path::PathBuf;
 
-// Link import cgo function
-#[link(name = "cgo")]
-extern "C" {
-    pub fn evaluateGo(policy: GoString, data: GoString, input: GoString) -> *mut c_char;
-}
-
-/// String structure passed into cgo
-#[derive(Debug)]
-#[repr(C)]
-pub struct GoString {
-    pub p: *const c_char,
-    pub n: isize,
-}
-
 #[derive(Debug, Clone)]
 pub struct Opa {
     policy_path: PathBuf,
 }
 
 impl Opa {
-    pub fn new(policy_path: PathBuf) -> Result<Self> {
+    pub fn new(policy_path: PathBuf) -> Result<Self, ResourcePolicyError> {
+        std::fs::create_dir_all(policy_path.parent().unwrap())
```

Comment:
is there a reason for assuming policy_path will never be / or should we return an Error?

Reply:
`create_dir_all` creates the path recursively if it does not already exist. It returns an error only if it tries to create the path and is unable to.